packages/mock-block-dock/src/datastore/mutate-subgraph.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  Entity,
+  EntityId,
+  EntityVertex,
+  KnowledgeGraphOutwardEdge,
+  KnowledgeGraphRootedEdges,
+  KnowledgeGraphVertices,
+  Subgraph,
+  Timestamp,
+} from "@blockprotocol/graph";
+import isEqual from "lodash/isEqual";
+
+/** @todo - clean up the assertions here */
+export const addKnowledgeGraphEdge = (
+  subgraph: Subgraph,
+  sourceEntityId: EntityId,
+  atTime: Timestamp,
+  outwardEdge: KnowledgeGraphOutwardEdge,
+) => {
+  /* eslint-disable no-param-reassign -- We want to mutate the input here */
+  if (!subgraph.edges[sourceEntityId]) {
+    // This is needed because ts can't differentiate between `EntityId` and `BaseUri`
+    (subgraph.edges as KnowledgeGraphRootedEdges)[sourceEntityId] = {
+      [atTime]: [outwardEdge],
+    };
+  } else if (!subgraph.edges[sourceEntityId]![atTime]) {
+    subgraph.edges[sourceEntityId]![atTime] = [outwardEdge];
+  } else {
+    const outwardEdgesAtTime = subgraph.edges[sourceEntityId]![atTime]!;
+    /**
+     * @todo - Q for PR review: Added a lodash dependency for this, equality of the outward edge is actually complicated
+     *    fine to keep?
+     */
```

Comment:
I saw our `util.ts` file had some patched utilities to avoid a heavy weight lodash dependency or something. I didn't see any for `isEqual`. I'd experimented with some other deep equality strategies but the types (lack of branding and such) makes it quite difficult to write it nicely.

Reply:
These are just objects, right? Simplest is this

```ts
const isObjectEqual = (
  first: Record<string, unknown>, 
  second: Record<string, unknown>
) => 
  JSON.stringify(first) === JSON.stringify(second)
```

Reply:
It's pretty messy but it'll almost certainly cover this case. I'll add it